### PR TITLE
Update Dockerfile

### PR DESCRIPTION
**Pull request by haf**
Fixes https://github.com/knative/eventing/issues/3011

Fixes #issue-number or description of the problem the PR solves

## Proposed Changes 

-
-
-

### CODE REVIEW

File: docs/eventing/samples/helloworld/helloworld-go/Dockerfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Use the offical Golang image to create a build artifact.
 # This is based on Debian and sets the GOPATH to /go.
 # https://hub.docker.com/_/golang
-FROM golang:1.12 as builder
+FROM golang:1.14 as builder
```
Comment (matzew) [inline docs/eventing/samples/helloworld/helloworld-go/Dockerfile:4]: I think we are on 1.13 ?

**Comment by sebgoa** [comment]
in addition to moving to Go 1.13, I think we should actually use `go mod init` instead of installing dep.

if we want to keep dep, we should use `go get` instead of fetching it via curl

File: docs/eventing/samples/helloworld/helloworld-go/Dockerfile (hunk 1 above)
Comment (akashrv) [reply docs/eventing/samples/helloworld/helloworld-go/Dockerfile:4]: Yes. This should be 1.13. @haf could you please change it and send an update. I'll LGTM.

Comment (sebgoa) [reply docs/eventing/samples/helloworld/helloworld-go/Dockerfile:4]: can we get rid off go dep as well ?

**Pull request closed without merging**